### ISSUE TITLE
in Tb128 - popup for restricted features: notification leads to exception + icon not displayed

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/QuickFolders#486]
This is caused by the function `NotificationBox.appendNotification()` having been converted to async in mozilla-central 128 (Firefox' codebase)

```
Uncaught (in promise) TypeError: newNotification.messageImage is undefined
    popupRestrictedFeature chrome://quickfolders/content/quickfolders-util.js:415
    onSkipFolder chrome://quickfolders/content/quickfolders-interface.js:541
    oncommand about:3pane:1
quickfolders-util.js:415:11
```

For example, when the user does something like "Skip unread folder", without a QF license, they get a notification such as the on shown below:

![image](https://github.com/user-attachments/assets/cf23c81d-05f2-458a-ba91-b77e6ec059d2)

The QuickFolders icon should also be displayed on the left of the notification. It is also possible that a function doesn't fully execute after the exception is thrown.

**Comment by RealRaven2000** [comment]
Here is a quick fix:

[QuickFolders-mx-6.6.2pre2.zip](https://github.com/user-attachments/files/16488407/QuickFolders-mx-6.6.2pre2.zip)

shows the correct notification and throws no more exceptions:

![image](https://github.com/user-attachments/assets/b6950a54-49ba-4b15-9601-b6951c160bec)


-----

To try the version above download zip file and drag the file into Thunderbird Add-ons Manager (do not extract contents, it won't install like that)

**Comment by RealRaven2000** [comment]
Implemented in 6.7, Published 17/08/2024
